src/core/server/app/middleware/passport/strategies/verifiers/sso.ts

```diff
@@ -60,8 +61,9 @@ export const SSOUserProfileSchema = Joi.object()
       .required(),
     username: Joi.string().required(),
     badges: Joi.array().items(Joi.string()),
+    role: Joi.string(),
```

Comment:
You can add a role specifier here:

```suggestion
    role: Joi.string().only(Object.values(GQLUSER_ROLE)),
```